### PR TITLE
Amplify Auth typescript enhancement

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
Add typescript definition to Auth module
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by powerful23** [comment]
@yuntuowang I changed some definition in amazon-cognito-identity-js, can you review that part?

File: packages/aws-amplify/src/Auth/types/Auth.ts [hunk 1]
```diff
@@ -17,9 +17,10 @@
 * Parameters for user sign up
 */
 export interface SignUpParams {
-    username: string;
-    password: string;
-    SignupAttributes?: Object;
+    username: string,
+    password: string,
+    attributes?: object,
+    validationData?: Array<any>,
```
Comment (yuntuowang) [inline packages/aws-amplify/src/Auth/types/Auth.ts]: Does the "validationData" mean the verification code from SMS or email? Why is it Array<any>?

Comment (powerful23) [reply packages/aws-amplify/src/Auth/types/Auth.ts]: Based on this line: https://github.com/aws/aws-amplify/blob/master/packages/amazon-cognito-identity-js/src/CognitoUserPool.js#L83

Comment (yuntuowang) [reply packages/aws-amplify/src/Auth/types/Auth.ts]: Got it. Can you change the type of it to CognitoUserAttribute[]?

Comment (powerful23) [reply packages/aws-amplify/src/Auth/types/Auth.ts]: Changed.

File: packages/aws-amplify/src/API/API.ts [hunk 2]
```diff
@@ -452,7 +452,7 @@ export default class APIClass {
                 const cred = Auth.essentialCredentials(credentials);
                 logger.debug('set credentials for api', cred);
```
Comment (jordanranz) [inline packages/aws-amplify/src/API/API.ts]: I'm confused with what the cred variable and call to Auth.essentialCredentials is doing. Is this setting credentials for api or just ensuring the credentials are returned properly? Is this call needed here?

File: packages/aws-amplify/__tests__/Auth/auth-unit-test.ts [hunk 3]
```diff
@@ -1199,7 +1203,7 @@ describe('auth unit test', () => {
             });
 
             expect.assertions(1);
-            expect(await auth.verifyUserAttribute(user, {})).toBe("success");
+            expect(await auth.verifyUserAttribute(user, {})).toBeUndefined();
```
Comment (jordanranz) [inline packages/aws-amplify/__tests__/Auth/auth-unit-test.ts]: What exactly is this testing?

File: packages/aws-amplify/src/API/API.ts (hunk 2 above)
Comment (jordanranz) [reply packages/aws-amplify/src/API/API.ts]: Ok, it is probably nice to have for the debug statement.

**Comment by jordanranz** [comment]
LGTM

**Pull request merged**